elements/order-input.html

```diff
@@ -67,9 +67,22 @@
                         if (header) header.hidden = true;
                     }
 
-                    var order = new parsers.OrderUpHtmlParser().parse(this.$.input);
-                    console.log('order', order);
-                    this._changeUrl(order);
+                    const parsersToTry = [parsers.OrderUpHtmlParser, parsers.OrderUpParser, parsers.CsvParser];
+                    const order = parsersToTry.reduce((acc, parser) => {
+                        if (acc && acc.people && acc.people.size > 0 && acc.total) {
```

Comment:
There's a `order.hasPeople` getter I've been using to check for empty orders.

Reply:
That isn't completely reliable, we'd still need the total check. If I type in a csv, it parses immediately and throws an error if no number has been entered yet (so "billy,"). The total check prevents the error. Also, I think throwing errors is rather annoying, especially in the constructor but also in the "split" function. We might want to consider another signifier of a bad order parsing

Reply:
BTW, I think that the constructor and `split()` should ultimately be the same method. There's no point in having to do them separately when you always have to do them one after the other.

Reply:
tl;dr: please get rid of deprecated `withPerson` `withTip` `withTax` invocations?